server/metrics/metrics.go

```diff
@@ -17,22 +17,29 @@ const (
 )
 
 type Metrics interface {
+	// Histogram metrics
 	ObserveAPIEndpointDuration(handler, method, statusCode string, elapsed float64)
-	ObserveConnectedUsersTotal(count int64)
+
+	// Counter metrics
+	IncrementHTTPRequests()
+	IncrementHTTPErrors()
 	ObserveChangeEventTotal(changeType string)
 	ObserveProcessedChangeEventTotal(changeType string, discardedReason string)
 	ObserveLifecycleEventTotal(lifecycleEventType string)
-	ObserveMessagesCount(action, source, isDirectMessage string)
-	ObserveReactionsCount(action, source, isDirectMessage string)
-	ObserveFilesCount(action, source, isDirectMessage, discardedReason string, count int64)
+	ObserveMessagesCount(action, source, isDirect string)
+	ObserveReactionsCount(action, source, isDirect string)
+	ObserveFilesCount(action, source, isDirect, discardedReason string, count int64)
+
+	// Gauge metrics
+	ObserveConnectedUsersTotal(count int64)
 	ObserveSyntheticUsersTotal(count int64)
 	ObserveLinkedChannelsTotal(count int64)
-	IncrementHTTPRequests()
-	IncrementHTTPErrors()
-	GetRegistry() *prometheus.Registry
+	ObserveUpstreamUsersTotal(count int64)
 	ObserveChangeEventQueueCapacity(count int64)
 	IncrementChangeEventQueueLength(changeType string)
 	DecrementChangeEventQueueLength(changeType string)
+
+	GetRegistry() *prometheus.Registry
```

Comment:
I know this was already in the interface, but wondering why we need `GetRegistry` exposed at all? This largely "breaks" the point of the interface if we intend to expose a concrete implementation detail.

Reply:
this was needed because we want the registry at this [part](https://github.com/mattermost/mattermost-plugin-msteams-sync/blob/423b7cc1f32dafb25850550b3bfe561a0d4e5008/server/metrics/server.go#L30) while defining the metrics server, so we defined an interface method to expose it.